zephyr/include/sof/drivers/timer.h

```diff
@@ -14,8 +14,18 @@
 struct comp_dev;
 struct sof_ipc_stream_posn;
 
-#define k_cycle_get_64_safe()		k_cycle_get_64()
-#define k_cycle_get_64_atomic()		k_cycle_get_64()
+#define CYCLES_PER_TICK (sys_clock_hw_cycles_per_sec() / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static inline uint64_t sof_cycle_get_64(void)
+{
+#if IS_ENABLED(CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER)
+	return k_cycle_get_64();
+#endif
+	return k_uptime_ticks() * CYCLES_PER_TICK;
```

Comment:
Better to use k_ticks_to_cyc_floor64() here. In this case I'm sure CYCLES_PER_TICK is an integer, but the Zephyr provided conversions are carefully designed to prevent overflows and other glitches with manual time math.